core/domain/exp_jobs_one_off_test.py

```diff
@@ -414,6 +416,183 @@ def test_nonhuman_committers_not_counted(self):
+                'cmd': 'edit_exploration_property',
+                'property_name': 'objective',
+                'new_value': 'New Objective'
+            }], 'Changed Objective.')
+
+        # Let the second user revert version 3 to version 2
+        exp_services.revert_exploration(user_b_id, self.EXP_ID, 3, 2)
+
+        # Run the job to compute the contributors summary.
+        job_id = exp_jobs_one_off.ExplorationContributorsSummaryOneOffJob.create_new()
+        exp_jobs_one_off.ExplorationContributorsSummaryOneOffJob.enqueue(job_id)
+        self.process_and_flush_pending_tasks()
+
+        # Check that the contributors_summary contains only user_a_id &
+        # not user_b_id
+        exploration_summary = exp_services.get_exploration_summary_by_id(
+            exploration.id)
+        self.assertIn(user_a_id, exploration_summary.contributors_summary)
+        self.assertNotIn(user_b_id, exploration_summary.contributors_summary)
+
+        # Check that the User A has only 2 commits after user b has reverted to version 2
+        self.assertEquals(2, exploration_summary.contributors_summary[user_a_id])
+
+    def test_reverts_not_counted(self):
+        """Test that if both non-revert commits and revert are
+        made by contributor then the contributions summary shows
+        only non-revert commits for that contributor. However,
+        the commits made after the version to which we have reverted
+        shouldn't be counted either.
+        """
+
+        user_a_id = self.get_user_id_from_email(self.EMAIL_A)
+
+        # Let USER A make 2 non-revert commits and 1 revert
+        exploration = self.save_new_valid_exploration(
+            self.EXP_ID, user_a_id, title='Exploration Title')
+        exp_services.update_exploration(
+            user_a_id, self.EXP_ID, [{
+                'cmd': 'edit_exploration_property',
+                'property_name': 'title',
+                'new_value': 'New Exploration Title'
+            }], 'Changed title.')
+        exp_services.update_exploration(
+            user_a_id, self.EXP_ID, [{
+                'cmd': 'edit_exploration_property',
+                'property_name': 'objective',
+                'new_value': 'New Objective'
+            }], 'Changed Objective.')
+
+        # Let USER A revert version 2 to version 1
+        exp_services.revert_exploration(user_a_id, self.EXP_ID, 3, 2)
+
+        # Run the job to compute the contributor summary.
+        job_id = exp_jobs_one_off.ExplorationContributorsSummaryOneOffJob.create_new()
+        exp_jobs_one_off.ExplorationContributorsSummaryOneOffJob.enqueue(job_id)
+        self.process_and_flush_pending_tasks()
+
+        # Check that USER A's number of contributions is equal to 2
+        exploration_summary = exp_services.get_exploration_summary_by_id(
+            exploration.id)
+        self.assertEqual(2, exploration_summary.contributors_summary[user_a_id])
+
+    def test_nonhuman_committers_not_counted(self):
```

Comment:
This method looks really good. Can we test all system ids, what do you think ? 
`for system_id in feconf.SYSTEM_USER_IDS
       exp_services.update_exploration(
           feconf.MIGRATION_BOT_USERNAME, self.EXP_ID, [{
                'cmd': 'edit_exploration_property',
                'property_name': 'title',
                'new_value': 'Title changed by ' + system_id
            }], 'Changed title.')
`
then you can run jobs. 
finally verify:

```
for system_id in feconf.SYSTEM_USER_IDS:
    self.assertNotIn(
            system_id,
             exploration_summary.contributors_summary)
```

This is not important suggestion. The main benefit is if in the future new system id will come up, this test will include it as well.